lib/Config.ts

```diff
@@ -176,18 +167,37 @@ class Config {
     this.dbpath = this.getDefaultDbPath();
   }
 
-  private updateMacaroonPaths = (network: string) => {
-    this.network = network as Network;
+  private updateNetwork = (args?: { [argName: string]: any }) => {
+    const networks: { [val: string]: boolean } = {
+      [XUNetwork.MainNet]: args!.mainnet,
+      [XUNetwork.TestNet]: args!.testnet,
+      [XUNetwork.RegTest]: args!.regtest,
+    };
+
+    const selected = Object.keys(networks).filter(key => networks[key]);
+    if (selected.length > 1) {
+      throw Error('only one alternative network selection is allowed');
+    }
+
+    if (selected.length === 0) {
+      this.network = XUNetwork.SimNet;
+    } else {
+      this.network = selected[0] as XUNetwork;
+    }
+  }
+
+  private updateMacaroonPaths = () => {
+    const lnNetwork = lnNetworks[this.network];
     for (const currency in this.lnd) {
       switch (currency) {
         case 'LTC':
           // litecoin uses a specific folder name for testnet
           this.lnd.LTC!.macaroonpath = path.join(this.lnd.LTC!.macaroonpath, '..', '..',
-            this.network === Network.TestNet ? 'testnet4' : this.network, 'admin.macaroon');
+            lnNetwork === XUNetwork.TestNet ? 'testnet4' : lnNetwork, 'admin.macaroon');
```

Comment:
I didn't leave it the way it was before because `lnNetwork` might be different from `xuNetwork` (for `simnet` and `regtest`). 
You're right about not comparing it with `XUNetwork`. To do so, i'll need to create another enum, to contain the two LN networks (`mainnet` / `testnet` only).
Let me know if you want me to do so.

Reply:
Why wouldn't comparing against `lnNetworks[XUNetwork.TestNet]` work? Then it'd be the same as line 95, right?